lib/feed.js

```diff
@@ -453,10 +453,11 @@ Feed.prototype._putRoots = function (top, batch, nodes, nodesOffset, sig, cb) {
 
     if (indexes.length) {
       var newBlocks = flat.rightSpan(indexes[indexes.length - 1]) / 2 + 1
-      if (newBlocks > self.blocks) {
+      var oldBlocks = self.blocks
+      if (newBlocks > oldBlocks) {
         self.blocks = newBlocks
         self.bytes = byteSize(roots)
-        self.emit('update')
+        while (oldBlocks++ < newBlocks) self.emit('update', oldBlocks)
```

Comment:
No worries! Like you said on IRC, one can just listen to update and keep track of before/after blocks. I'll close here :)